### PR TITLE
Feat/sso account recovery

### PR DESCRIPTION
**Pull request by TheLeoB**
## Description (optional)
1. Adds modal in wallet in case of account reset failure
2. Add feature flag to switch between old and new magic login email endpoints
3. Adds link to reset 2fa on 2fa input form
4. Design feedback

## Testing Steps (optional)
Detail the steps required for the reviewer(s) to verify and test these changes.

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts [hunk 1]
```diff
@@ -718,13 +718,20 @@ export default ({ api, coreSagas, networks }) => {
   const triggerWalletMagicLink = function* (action) {
     const formValues = yield select(selectors.form.getFormValues('login'))
     const { step } = formValues
+    const legacyMagicEmailLink = (yield select(
+      selectors.core.walletOptions.getFeatureLegacyMagicEmailLink
+    )).getOrElse(false)
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/auth/sagas.ts:723]: i think we want to default to true right? `getOrElse(true)`. at least for the timebeing

File: packages/blockchain-wallet-v4-frontend/src/modals/Wallet/ResetAccountFailed/index.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+          </Text>
+          <RectangleBackground>
+            <Text size='12px' color='grey600' weight={500}>
+              <FormattedMessage
+                defaultMessage='Fund recovery failures can happen for a number of reasons. Our support team is able to help recover your account. <a>Learn more</a>'
+                id='modals.resetaccountfailure.information'
+                values={{
+                  a: (msg) => (
+                    <a
+                      href='https://www.figma.com/exit?url=https%3A%2F%2Fsupport.blockchain.com%2Fhc%2Fen-us%2Farticles%2F4404679303700'
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/modals/Wallet/ResetAccountFailed/index.tsx:79]: do we have a real url for this?

**Review by schnogz — CHANGES_REQUESTED** [review]
looks really good overall!

File: packages/blockchain-wallet-v4-frontend/src/modals/Wallet/ResetAccountFailed/index.tsx (hunk 2 above)
Comment (TheLeoB) [reply packages/blockchain-wallet-v4-frontend/src/modals/Wallet/ResetAccountFailed/index.tsx:79]: 🤦‍♀️

**Review by schnogz — APPROVED** [review]

**Pull request merged**